src/classes/OPP_OpportunityNaming.cls

```diff
@@ -410,7 +410,11 @@ public with sharing class OPP_OpportunityNaming implements OPP_INaming {
             }
         } else if (dt == Schema.DisplayType.Currency) {
             if (obj.get(fieldName.trim())!=null) {
-                result = Label.npe01.DefaultContactTransactionCurrency + String.valueOf(obj.get(fieldName.trim()));
+                String currencySymbol = UTIL_Currency.getInstance().getCurrencySymbol((Id)obj.get('Id'));
+                if (currencySymbol.length() == 3) {
```

Comment:
In multicurrency orgs, it's always an ISO code that's attached to the amount, e.g. USD 500, so it needs the space. In single currency orgs it maps to a currency symbol, e.g. $500.